src/main/scala/edu/ie3/powerFactory2psdm/converter/NodeConverter.scala

```diff
@@ -8,95 +8,61 @@ package edu.ie3.powerFactory2psdm.converter
 
 import edu.ie3.datamodel.models.OperationTime
 import edu.ie3.datamodel.models.input.{NodeInput, OperatorInput}
-import edu.ie3.powerFactory2psdm.exception.pf.{
-  ElementConfigurationException,
-  GridConfigurationException
-}
+import edu.ie3.powerFactory2psdm.exception.pf.GridConfigurationException
 import edu.ie3.powerFactory2psdm.model.Subnet
-import edu.ie3.powerFactory2psdm.model.powerfactory.PowerFactoryGrid.{
-  ConElms,
-  Nodes
-}
+import edu.ie3.powerFactory2psdm.model.powerfactory.Node
 import edu.ie3.util.quantities.PowerSystemUnits.PU
 import org.locationtech.jts.geom.Point
 import tech.units.indriya.quantity.Quantities
-
 import java.util.UUID
-import scala.util.{Failure, Success, Try}
 
 object NodeConverter {
 
-  /**
-    * Checks if a node is a slack node by checking if there is an external grid connected to the node.
-    *
-    * @param maybeConElms an optional List of connected elements to the node
-    * @return either whether the node is a slack node or an exception
-    */
-  def isSlack(maybeConElms: Option[List[Option[ConElms]]]): Try[Boolean] =
-    maybeConElms match {
-      case Some(conElms) =>
-        conElms.flatten.filter(
-          conElm => conElm.pfCls.getOrElse("NO_CLASS_DEFINED") == "ElmXnet"
-        ) match {
-          case Seq(_) => Success(true)
-          case Nil    => Success(false)
-          case _ =>
-            Failure(
-              GridConfigurationException(
-                "There is more than one external grid connected to the node."
-              )
-            )
-        }
-      case None =>
-        Failure(
-          ElementConfigurationException(
-            "The optional connected elements attribute is None."
-          )
-        )
-    }
-
   /**
     * Converts a PowerFactory node into a PSDM node.
     *
-    * @param nodeUUID  UUID of the PF node to convert
-    * @param UUID2node Map of UUIDs and their associated [[Nodes]]
+    * @param id  id of the PF node to convert
+    * @param id2node Map of ids and their associated [[Node]]s
     * @param subnet    subnet the PF node lives in
     * @return a PSDM [[NodeInput]]
     */
   def convertNode(
-      nodeUUID: UUID,
-      UUID2node: Map[UUID, Nodes],
+      id: String,
+      id2node: Map[String, Node],
       subnet: Subnet
   ): NodeInput = {
-    val pfNode = UUID2node(nodeUUID)
+    val node = id2node(id)
```

Comment:
Unsafe! Throws Exception if `id` is not in map. 
I know that this is impossible as of `GridConverter:35` but its not clear visible at this point and if somebody modifies `GridConverter:35` in the future, this might explode here ... -> not good 

Additionally, you have duplicate data here (`node` contains its `id`) 

Suggestion: 

```
  def convertNode(
      node: Node,
      subnet: Subnet
  ): NodeInput = {
    val subnetId = subnet.id
    val vTarget = Quantities.getQuantity(node.vTarget, PU)
    val geoPosition: Point = CoordinateConverter.convert(node.lat, node.lon)
    val voltLvl = subnet.voltLvl
    val slack = isSlack(node)
    new NodeInput(
      UUID.randomUUID(),
      node.id,
      OperatorInput.NO_OPERATOR_ASSIGNED,
      OperationTime.notLimited(),
      vTarget,
      slack,
      geoPosition,
      voltLvl,
      subnetId
    )
  }
```